params.go

```diff
@@ -597,29 +624,56 @@ func (p *Params) ValidateRequiredProperties() (bool, []error) {
 		}
 	}
 
-	// validate sidecar params
-	if p.Sidecar.Type == "" {
+	// The "sidecar" field is deprecated, so it can be empty. But if it's specified, then we validate it.
+	if p.Sidecar.Type != "" && p.Sidecar.Type != "none" {
+		errors = validateSidecar(p.Sidecar, errors)
+	}
+
+	// validate sidecars params
+	for _, sidecar := range p.Sidecars {
+		errors = validateSidecar(sidecar, errors)
+	}
+
+	// Either the Sidecar has to be not empty, or Sidecars has to contain an element
+	if (p.Sidecar.Type == "" || p.Sidecar.Type == "none") && len(p.Sidecars) == 0 {
+		errors = append(errors, fmt.Errorf("Specifying a sidecar is required; Add one to the sidecars collection"))
```

Comment:
Can you list potential values for the type? I'm not sure why I thought a sidecar had to be required though, if you'd like to run without it's up to yourself. Checking if a type is valid option does make sense though.